src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -427,6 +427,19 @@ private TestAssemblyInfo GetAssemblyInfo(Type type)
                 if (IsAssemblyOrClassInitializeMethod<AssemblyInitializeAttribute>(methodInfo))
                 {
                     assemblyInfo.AssemblyInitializeMethod = methodInfo;
+
+                    if (_reflectionHelper.IsAttributeDefined<TimeoutAttribute>(methodInfo, false))
```

Comment:
Is it doing any optimization to use `IsAttributeDefined` instead of doing null check on `GetAttribute`? I am asking because if you are using the `if (GetXX is {} timeoutAttribute)` then you can skip the debug assert.

Reply:
`IsAttributeDefined` goes to the cache